runtime/src/installed_scheduler_pool.rs

```diff
@@ -23,14 +27,40 @@ use {mockall::automock, qualifier_attr::qualifiers};
     allow(unused_attributes, clippy::needless_lifetimes)
 )]
 pub trait InstalledScheduler: Send + Sync + Debug + 'static {
+    // Calling this is illegal as soon as wait_for_termination is called.
     fn schedule_execution<'a>(
         &'a self,
         transaction_with_index: &'a (&'a SanitizedTransaction, usize),
     );
+
+    #[must_use]
+    fn wait_for_termination(&mut self, reason: &WaitReason) -> Option<ResultWithTimings>;
 }
 
 pub type DefaultInstalledSchedulerBox = Box<dyn InstalledScheduler>;
 
+pub type ResultWithTimings = (Result<()>, ExecuteTimings);
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum WaitReason {
```

Comment:
Could you also comment on this from the call-side perspective? It's unclear to me, when I call `wait_for_terminate` with one of these reasons...is that me telling the scheduler to terminate with that reason, or is it just waiting for the reason to happen?

Reply:
did my best: 507c5023f35b4b435c5fce9493c345ed924f9c60